### ISSUE TITLE
Version 1.5.2 on Rails 6.0.1 causing deprecation warning

### ISSUE THREAD

**Issue by givemetraffic** [github/actionview-component#135]
Rails 6.0.1
Gem version 1.5.2

In config/application.rb,
`require "action_view/component"`

Causing: 
```
DEPRECATION WARNING: Initialization autoloaded the constants ActionText::ContentHelper and ActionText::TagHelper.

Being able to do this is deprecated. Autoloading during initialization is going
to be an error condition in future versions of Rails.

Reloading does not reboot the application, and therefore code executed during
initialization does not run again. So, if you reload ActionText::ContentHelper, for example,
the expected changes won't be reflected in that stale Module object.

These autoloaded constants have been unloaded.

Please, check the "Autoloading and Reloading Constants" guide for solutions.
 (called from <main> at /Users/
USER/Projects/rails-app-test/config/environment.rb:5)
```
at `development.log`

Which is highly difficult to track

Related issue https://github.com/rails/rails/issues/36546

**Comment by givemetraffic** [comment]
Version 1.4.0 works ok, though

In `config/application.rb`,
require "action_view/component/base"

**Comment by richardkmiller** [comment]
For me, v1.5.3 also works if I change `require "action_view/component"` to `require "action_view/component/base"`.

**Comment by givemetraffic** [comment]
`require "action_view/component/base"` doesn't work with 1.5.3 though -- it's causing something like `'ExampleComponent' is not an ActiveModel-compatible object. It must implement :to_partial_path.`

#79

**Comment by johanandre** [comment]
I'm using Rails 6.0.2.1 and latest release of AVC and getting the deprecation warning too... Any ideas of how to get rid of it?

**Comment by joelhawksley** [comment]
@johanandre we've yet to have someone look at reproducing / fixing this issue. Might you be interested in taking a crack at it?

**Comment by kinduff** [comment]
I was able to track it down to this line: https://github.com/github/actionview-component/blob/22068157e0083589cf350407aa80473e85926ecd/lib/railties/lib/rails/components_controller.rb#L3. Based on this [documentation](https://guides.rubyonrails.org/autoloading_and_reloading_constants.html#introduction) , this should be avoided.

I'm taking a shot of fixing this warning, I'll post my progress soon.

**Comment by joelhawksley** [comment]
Thanks @kinduff! Please let us know what you find, even if it's not a complete fix ❤️

**Comment by skahlert** [comment]
@kinduff if I can support you, let me know! Also have this issue in my setup.

**Comment by joelhawksley** [comment]
This should be fixed in 1.12.0, by @seanpdoyle's https://github.com/github/actionview-component/pull/225. Please re-open if not! ❤️
